sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -0,0 +1,437 @@
+     *
+     * @return A reactive response containing the file data.
+     */
+    public Flux<ByteBuffer> read() {
+        try {
+            return readWithResponse(null, null, null, false)
+                .flatMapMany(FileDownloadAsyncResponse::getValue);
+        } catch (RuntimeException ex) {
+            return fluxError(logger, ex);
+        }
+    }
+
+    /**
+     * Reads a range of bytes from a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.readWithResponse#FileRange-DownloadRetryOptions-DataLakeRequestConditions-boolean}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob">Azure Docs</a></p>
+     *
+     * @param range {@link FileRange}
+     * @param options {@link DownloadRetryOptions}
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @param rangeGetContentMD5 Whether the contentMD5 for the specified file range should be returned.
+     * @return A reactive response containing the file data.
+     */
+    public Mono<FileDownloadAsyncResponse> readWithResponse(FileRange range, DownloadRetryOptions options,
+        DataLakeRequestConditions accessConditions, boolean rangeGetContentMD5) {
+        try {
+            return blockBlobAsyncClient.downloadWithResponse(Transforms.toBlobRange(range),
+                Transforms.toBlobDownloadRetryOptions(options), Transforms.toBlobRequestConditions(accessConditions),
+                rangeGetContentMD5).map(response -> new FileDownloadAsyncResponse(response.getRequest(),
+                response.getStatusCode(), response.getHeaders(), response.getValue(),
+                response.getDeserializedHeaders()));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Moves the file to another location within the file system.
+     * For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create">Azure
+     * Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.rename#String}
+     *
+     * @param destinationPath Relative path from the file system to rename the file to.
+     * @return A {@link Mono} containing a {@link DataLakeFileAsyncClient} used to interact with the new file created.
+     */
+    public Mono<DataLakeFileAsyncClient> rename(String destinationPath) {
+        try {
+            return renameWithResponse(destinationPath, null, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Creates a new sub-directory within a directory. If a sub-directory with the same name already exists, the
```

Comment:
>sub-directory [](start = 21, length = 13)

Creates a new file

Reply:
I'll add an issue for this

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6104